backend/app/usecase/repository/userurl_fake.go

```diff
@@ -6,14 +6,32 @@ var _ UserURLRelation = (*UserURLRelationFake)(nil)
 
 // UserURLRelationFake represents in memory implementation of User-URL relationship accessor.
 type UserURLRelationFake struct {
+	userURLRelations map[string]string
 }
 
 // CreateRelation creates many to many relationship between User and URL.
 func (u UserURLRelationFake) CreateRelation(user entity.User, url entity.URL) error {
+	if u.userURLRelations == nil {
+		u.userURLRelations = make(map[string]string)
+	}
+	u.userURLRelations[url.Alias] = user.Email
 	return nil
 }
 
+// FindAliasesByUser fetches all the aliases corresponding to the given user.
+func (u UserURLRelationFake) FindAliasesByUser(user entity.User) ([]string, error) {
+	var aliases []string
+	for alias := range u.userURLRelations {
+		if u.userURLRelations[alias] == user.Email {
+			aliases = append(aliases, alias)
+		}
+	}
+	return aliases, nil
+}
+
 // NewUserURLRepoFake creates UserURLFake
-func NewUserURLRepoFake() UserURLRelationFake {
-	return UserURLRelationFake{}
+func NewUserURLRepoFake(userURLRelations map[string]string) UserURLRelationFake {
```

Comment:
```suggestion
func NewUserURLRepoFake(userURLRelations map[string]string) (UserURLRelationFake, error) {
```
Let's do error handling up front here to save us from handling error on every API call.